### PR TITLE
linedash support

### PR DESCRIPTION
**Pull request by rvilarl**
intial step to support #157 

it also reesolves the deep copy in svgcontext.

no visual changes

### CODE REVIEW

**Comment by rvilarl** [comment]
I have included the changes necessary to support dotted lines in curves. And modified two test to show the result. These are the only visual differences:

current
svg_Curve Rounded_Curve Bravura svg_current
reference
svg_Curve Rounded_Curve Bravura svg_reference
current
svg_Curve Simple_Curve Bravura svg_current
reference
svg_Curve Simple_Curve Bravura svg_reference

**Comment by ronyeh** [comment]
The visual output looks good to me for now.

For reference, this is how Dorico does it:
https://steinberg.help/dorico/v1/en/dorico/topics/notation_reference/notation_reference_slurs_styles_r.html

I think that the ideal end result will be to have the dotted / dashed slur be thicker in the middle and taper at both ends (like the normal solid slur curve).

To do that, we might have to implement it as a filled shape, rather than a path with a dashed stroke.

But for now, we can do it this way.

File: src/svgcontext.ts [hunk 1]
```diff
@@ -104,10 +100,9 @@ export class SVGContext extends RenderContext {
 
     this.path = '';
     this.pen = { x: NaN, y: NaN };
-    this.lineWidth = 1.0;
 
     const defaultFontAttributes = {
-      'font-family': Metrics.get('fontFamily'),
+      'font-family': Metrics.get('fontFamily') as string,
```
Comment (ronyeh) [inline src/svgcontext.ts:105]: Curious why this cast was needed? Is it not normally a string?

**Review by ronyeh — APPROVED** [review]

File: src/svgcontext.ts (hunk 1 above)
Comment (rvilarl) [reply src/svgcontext.ts:105]: get returns any :(

**Pull request merged**